src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs

```diff
@@ -34,13 +35,20 @@ public void WriteExtension(CodeWriter writer, BuildContext<MgmtOutputLibrary> co
                 {
                     foreach (var resource in resources)
                     {
-                        if (ParentDetection.ParentResourceType(resource.OperationGroup, context.Configuration.MgmtConfiguration).Equals(ResourceTypeBuilder.Subscriptions))
+                        if (!resource.OperationGroup.IsSingletonResource(context.Configuration.MgmtConfiguration)&&
+                            ParentDetection.ParentResourceType(resource.OperationGroup, context.Configuration.MgmtConfiguration).Equals(ResourceTypeBuilder.Subscriptions))
```

Comment:
move this to an outer if statement and make the is singleton an inner if statement